mlflow/utils/rest_utils.py

```diff
@@ -133,7 +134,13 @@ def http_request(
 
     from mlflow.tracking.request_header.registry import resolve_request_headers
 
-    headers = dict({**_DEFAULT_HEADERS, **resolve_request_headers()})
+    request_headers = resolve_request_headers()
```

Comment:
Thanks @dbczumar I missed this part. My suggestion would be yes to concatenate other headers, since if multiple providers are adding same header, if header added by one provider A gets overwritten by provider B, it is equivalent to not having provider B.

Reply:
@dbczumar  I have update the PR to concatenate header with same name. Please review if it is a valid change to amke and share your comments. Thanks